### PR TITLE
Karma: puppeeter support and native events

### PR DESCRIPTION
**Pull request by dvoytenko**
Forked from https://github.com/google/web-stories-wp/pull/1712

Partial for #1721.

Blocked by #1695.

### CODE REVIEW

File: karma.config.js [hunk 1]
```diff
@@ -98,7 +107,14 @@ module.exports = function (config) {
 
     // start these browsers
     // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['Chrome'],
+    browsers: ['puppeteer'],
+
+    puppeteerLauncher: {
+      puppeteer: {
+        headless: config.headless || false,
```
Comment (cvolzke4) [inline karma.config.js:114]: [nit]: could this be just:

headless: Boolean(config.headless) or
headless: !!config.headless

File: karma/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Karma integration testing
+
+## Browser support
+
+Currently, only Puppeeter option is configured. It can run Chrome and Firefox browsers.
+In the near future this may also be expanded to Playwright to support Safari and other
```
Comment (cvolzke4) [inline karma/README.md:6]: Puppeteer will also support Safari eventually :) but nobody knows when.

File: karma/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Karma integration testing
+
+## Browser support
+
+Currently, only Puppeeter option is configured. It can run Chrome and Firefox browsers.
```
Comment (cvolzke4) [inline karma/README.md:5]: [nit]: Puppeteer

File: karma/karma-puppeteer-client/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const path = require('path');
+
+function createPattern(pattern) {
+  return { pattern: pattern, included: true, served: true, watched: false };
```
Comment (cvolzke4) [inline karma/karma-puppeteer-client/index.js]: [nit]: { pattern, included: true...

File: .eslintrc [hunk 5]
```diff
@@ -200,7 +200,17 @@
...
     {
       "files": [
-        "**/karma/**/*.js",
+        "**/karma/**/*.js"
+      ],
+      "plugins": [
+        "header"
+      ],
+      "rules": {
+        "no-undef": "off"
```
Comment (wassgha) [inline .eslintrc]: Any reason why?

File: karma/karma-puppeteer-launcher/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+The available options are currently exposed via Karma command line args:
+
+- `slowMo`: Slows down Puppeteer operations by the specified amount of milliseconds. Default is `0`.
+- `headless`: Runs in headless mode. Default is `false`.
+
+For instance, you can run Karma with the following options:
+
+```sh
+npm run test:karma --headless --slowMo=100
```
Comment (wassgha) [inline karma/karma-puppeteer-launcher/README.md:17]: nit
```suggestion
npm run test:karma -- --headless --slowMo=100
```

**Review by wassgha — APPROVED** [review]
Looks great!

File: karma/README.md (hunk 2 above)
Comment (dvoytenko) [reply karma/README.md:6]: Sounds good.

File: .eslintrc (hunk 5 above)
Comment (dvoytenko) [reply .eslintrc]: These are files specific to Karma configuration and there are a number of Jasmine and Puppeteer globals and I figured it's ok to just turn them off for now.

Comment (swissspidy) [reply .eslintrc]: Let's use

```js
{
  "env": {
    "jasmine": true
  }
}
```

instead

Comment (dvoytenko) [reply .eslintrc]: That's a lot better! Done.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**